### PR TITLE
test(html): improve test coverage for html interfaces

### PR DESCRIPTION
**Pull request by pgoldrbx**
## Description

- covers more of the common use cases
- edge cases for old browser support are still not currently covered by tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore, documentation, cleanup

## Motivation and Context

Attempting to improve coverage before upgrading `jsdom` (#20)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have updated the documentation (if required).
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/html.test.js [hunk 1]
```diff
@@ -89,6 +89,117 @@ describe('html', function() {
...
+        const elem = document.createElement('input');
+        // NOTE: JsonML style attribute must provide the literal inline CSS string value.
+        // An object of DOM style properties (e.g. marginTop) will not be applied as expected.
+        const attrs = {
+          style: 'color: blue; margin-top: 25px;',
+        };
+        const jml = ['', attrs];
+        html.patch(elem, jml);
+        assert.equal(elem.outerHTML, `<input style="${attrs.style}">`);
+      });
```
Comment (pgoldrbx) [inline test/html.test.js]: I don't know why, but I initially expected the library to nicely handle a DOM style object. I guess that would be a nice feature in the future.

File: test/html.test.js [hunk 2]
```diff
@@ -89,6 +89,117 @@ describe('html', function() {
...
+
+      it('should be able to update set event handlers through attributes', function(done) {
+        const elem = document.createElement('input');
+        // NOTE: event names must be lowercase. camelCase will currently fail
+        // click to complete the test otherwise the test will fail due to timeout
+        html.patch(elem, ['', { onclick: () => done() }]);
+        elem.dispatchEvent(new document.defaultView.Event('click'));
```
Comment (pgoldrbx) [inline test/html.test.js]: Another future patch perhaps. `onClick` will not work. A simple `toLowerCase()` would probably do a world of good.

File: test/html.test.js [hunk 3]
```diff
@@ -89,6 +89,117 @@ describe('html', function() {
...
+
+    describe('tables', function() {
+      it('should attempt to append a TD cell to by creating a tbody', function() {
+        const elem = document.createElement('table');
+        const jml = ['',
+          ['td', 'rogue table cell'],
+        ];
+        html.patch(elem, jml);
+        assert.equal(elem.outerHTML, '<table><tbody><td>rogue table cell</td></tbody></table>');
+      });
```
Comment (pgoldrbx) [inline test/html.test.js]: The code that supports this is a bit odd. It's buggy as it stands. The `tbody` should also include a row. However, it would be simpler to just disallow (throw an error) appending an orphaned table cell to a table without any idea of where it ought to go.

Still, I've added tests based on the current behavior to help illustrate what the library is currently doing, if nothing else. I'm not sure what to do about it other than immediately file one or more issues.

File: test/html.test.js [hunk 4]
```diff
@@ -150,6 +261,185 @@ describe('html', function() {
...
+        assert.equal(elem.outerHTML, '<input disabled="">');
+      });
+
+      it('should coerce date objects to strings setting attribute values', function() {
+        const someDate = new Date('2015-06-15');
+        const jml = ['div', {
+          'data-date': someDate,
+        }];
+        const elem = html.toHTML(jml);
+        assert.equal(elem.outerHTML, `<div data-date="${someDate.toString()}"></div>`);
```
Comment (pgoldrbx) [inline test/html.test.js]: This is helpful to demonstrate what happens. If used for the value of `<time datetime={...}` you'll find that the format is not as desired and it will be better to provide a string.

File: test/html.test.js [hunk 5]
```diff
@@ -150,6 +261,185 @@ describe('html', function() {
...
+      // The code branch inserts a space after the message
+      it('should create a shorthand comment from JsonML', function() {
+        const message = 'this is a comment';
+        const jml = [`! ${message}`];
+        const elem = html.toHTML(jml);
+        assert.equal(elem.toString(), '[object Comment]');
+        const div = document.createElement('div');
+        div.appendChild(elem);
+        assert.equal(div.innerHTML, `<!-- ${message} -->`);
+      });
```
Comment (pgoldrbx) [inline test/html.test.js]: This was an odd discovery. I don't really know why it exists.

**Comment by pgoldrbx** [comment]
I was able to improve coverage a little bit more for some uncommon attribute cases.

Overall for the library, after this PR:
```
----------|---------|----------|---------|---------|
File      | % Stmts | % Branch | % Funcs | % Lines |
----------|---------|----------|---------|---------|
All files |   85.98 |    74.77 |   91.67 |   85.87 | 
```

And I feel much more confident about the `jsdom` update.

However, I'm still not totally sure what to do about some of the oddities I found while working on the tests in this PR.

File: test/html.test.js (hunk 3 above)
Comment (pgoldrbx) [reply test/html.test.js]: Filed https://github.com/CondeNast/jsonml.js/issues/27 to handle these
